templates/CRM/Member/Form/MembershipPriceSetCommon.tpl

```diff
@@ -0,0 +1,10 @@
+{if $priceSetId}
+  {include file="CRM/Price/Form/PriceSet.tpl" context="standalone" extends="Membership"}
+  <script>
+    {if $price_set_has_new_orgs}
```

Comment:
I don't quite follow what @eileenmcnaughton means here. My skim on what is happening is that if the price set includes one or more organizations that the contact does not have a membership in (or current membership? or current membership that has not expired?) then you want to show a Join date description field. But I think the description is more confusing than enlightening, so I'd also recommend getting rid of this join date description show/hide stuff.